### DISCUSSION TITLE
Displaying author and date in content

### DISCUSSION THREAD

**Discussion by jehoshua7** [McShelby/hugo-theme-relearn#69] [Q&A] [answered]
If an .md file has

`author: Donald Duck
date: 2020-02-24`

specified, how can this be displayed in the HTML ?  I can see that one of the showcases for the **learn** theme has it shown/displayed, see https://tshark.dev/formats/format_usage/

Have read through all the documentation for the theme and can't find out how to turn this feature on ?

**Comment by jehoshua7** [comment]
@McShelby has been giving me some tips in the thread at https://discourse.gohugo.io/t/displaying-author-and-date-in-content/34505 . This works okay

`<div class="page-description">
<i class="fas fa-clock"></i> {{ .ReadingTime }} min | <i class="fas fa-feather-alt"></i> {{ .Params.Author }} | <i class="far fa-calendar-check"></i> {{ .Date.Format "January 1, 2006"}}</div>

{{- .Content -}}`

It no doubt needs some conditional code; not everyone wants the date/time displayed.

(Not all the code is appearing in this Github post ; see https://discourse.gohugo.io/t/displaying-author-and-date-in-content/34505/7?u=jehoshua7 for all the code)

**Comment by jehoshua7** [comment]
Can this be added as an enhancement to the theme ?  A parameter/boolean to indicate whether to display the information or not. It's working fine on a live site
![Donald_duck](https://user-images.githubusercontent.com/33979036/131584192-b5bfa05b-113d-4048-8167-30d95a4c1da6.png)
 (see attachment)

**Comment by McShelby** [comment]
Although we have a similar footer at my companies site, usually every use case is different and requires different information, ordering and styling in the footer. This permutation can not be handled by the theme itself.

**Comment by jehoshua7** [comment]
> Although we have a similar footer at my companies site, usually every use case is different and requires different information, ordering and styling in the footer. 

This isn't in the footer though, it is a modification to `layouts/partials/content.html ` As there are already paramaters/boolean values setup in the theme I see no reason why not to add this enhancement. For example `disableShortcutsTitle = true` is a config value that simply controls the displaying of information, example code `<h3>{{ if not $disableShortcutsTitle}}{{ T "Shortcuts-Title"}}{{ end }}</h3>` , which is in `/layouts/partials/menu.html`

> This permutation can not be handled by the theme itself.

Yet the [learn theme](https://github.com/matcornic/hugo-theme-learn) that this theme was forked from, has a minor modification to display the read time, author and date. See https://tshark.dev/formats/pcapng_format/

Simply meaning it can be done and as [the example code](https://discourse.gohugo.io/t/displaying-author-and-date-in-content/34505/7 ) shows, it is only minor modifications.

All that said, I recognise that as you are the maintainer of this theme, I may not be able to persuade you though.   :+1: 

There was some mention of bringing in a templating system for content. Is that in place or in a future release please ?

Thanks for your help.  :)

**Reply by McShelby** [reply]
Just for terminologie: Although this proposed modification may not be part of the `footer.html` but instead it may be shown above the content, for the sake of simplicity and lack of a better word I have called it a footer. Let's clarify this and call it **content meta**.

> Yet the learn theme that this theme was forked from, has a minor modification to display the read time, author and date.

It is definetly possible, that's for sure! But how should this be implemented in a versatile way: Some want to have it above the content, some below, some above and below, depending on the meta information. Some want it left aligned, some to the right and other in the middle. This would require at least four new config options. 

And that's only the options for positioning. How to configure the displayed data itself? Some want only the author, the last modified timestamp, the reading time, some other arbitray date from the TOML section, etc.

And what about the ordering of that information? This must be configurable, too (as this would be the first ticket to be opened after the feature would be released without ordering).

> For example `disableShortcutsTitle = true` is a config value that simply controls the displaying of information

As you may see from my above explaination, your example is much simpler as it misses permutation. It's a simple on/off switch while the content meta feature would require much more configuration to be versatile. 

This is **a lot** of work with very low impact (at the moment I am still busy fixing outstanding bugs). If you want to work on this and provide a PR, I am willing to include this into the official release **if it provides the above mentioned flexiblity**.

At the moment a custom `content.html` is the best way to implement this into your installation with the downside of providing necessary code on your own.

**Comment by jehoshua7** [comment]
Thanks for your replies @McShelby , much appreciated.

> At the moment a custom content.html is the best way to implement this into your installation with the downside of providing necessary code on your own.

Yes it seems the best option all around. Will the content template concepts be an easier method in the long run ?   I'm just mindful of re-applying code modifications when I want/need to use an update of the theme.

**Comment by McShelby** [comment, marked as answer]
> Will the content template concepts be an easier method in the long run ? I'm just mindful of re-applying code modifications when I want/need to use an update of the theme.

Providing your changes in `layouts/partials/content.html` will survive all updates. Unless you're not relying on CSS of the theme, your layout will still look fine and will not need any changes after an update.

**Comment by jehoshua7** [comment]
> Providing your changes in layouts/partials/content.html will survive all updates

I use Beyond Compare to quickly check on any changes/additions, etc, so it should be very minimal indeed.

Thanks  :+1:

**Reply by McShelby** [reply]
You'll don't need to diff it, as this file is not contained in the release. You will create your own if you need changes. See [the Hugo documentation](https://gohugo.io/templates/lookup-order/#hugo-layouts-lookup-rules-with-theme)
